distribution/kernel-debuginfo/runtest.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+set -x
+
+. /usr/bin/rhts_environment.sh
+
+function test_pass()
+{
+	report_result $TEST PASS 0
+}
+
+function test_fail()
+{
+	report_result $TEST FAIL 1
+}
+
+function nvr_kernel_current()
+{
+	rpm -qa "kernel*" | grep "$(uname -r)" | sort | head -n1
+}
+
+function nvr_kernel_latest()
+{
+	yum list --available kernel{,-rt}.$(uname -m) --showduplicates \
+	| sed "s/.$(uname -m)//" \
+	| sort -k1,1 -k2V \
+	| awk 'END { print $1"-"$2".'$(uname -m)'"; }'
+}
+
+# Determine which kernel debuginfo to download.
+# By default, use currently running kernel (unless overriden from env).
+# This can be achieved either by specifying KERNEL_NVR directly, or one
+# of the following two test config options:
+# CONFIG_NVR_KERNEL_{CURRENT,LATEST}
+
+if [ -z "${KERNEL_NVR++}" ]
+then
+	echo "DEBUG: Using current kernel nvr."
+	KERNEL_NVR="$(nvr_kernel_current)"
+else
+	echo "DEBUG: Using user provided kernel NVR: $KERNEL_NVR"
+fi
+
+if [ -n "$CONFIG_NVR_KERNEL_CURRENT" ]
+then
+	echo "DEBUG: Using current kernel nvr."
+	KERNEL_NVR="$(nvr_kernel_current)"
+fi
+
+if [ -n "$CONFIG_NVR_KERNEL_LATEST" ]
+then
+	echo "DEBUG: Using latest available kernel."
+	KERNEL_NVR="$(nvr_kernel_latest)"
+fi
+
+if [ -z "${KERNEL_NVR++}" ]
+then
+	echo "ERROR: Unable to determine kernel nvr."
+	echo "DEBUG: CONFIG_NVR_KERNEL_CURRENT: " \
+	     $CONFIG_NVR_KERNEL_CURRENT
+	echo "DEBUG: CONFIG_NVR_KERNEL_LATEST: " \
+	     $CONFIG_NVR_KERNEL_LATEST
+	echo "DEBUG: rpm -qa \"kernel*\""
+	rpm -qa "kernel*"
+	echo "DEBUG: uname -r: $(uname -r)"
+	echo "DEBUG: yum list --available kernel{,rt}.$(uname -m)" \
+	     "--showduplicates:"
+	yum list --available kernel{,-rt}.$(uname -m) --showduplicates
```

Comment:
I'm guessing this test should be only ran for RHEL pipelines and not Fedora? We do test RPM builds from Koji but the moment we'd update to fc31 yum will no longer be symlinked so this would break, unless you can make it work for dnf too (and picking the command based on which one is available).

Reply:
In principle, the test makes sense for both. Regarding yum, I can just use something like:
```
YUM=$(command -v yum)
[[ -z $YUM ]] && YUM=$(command -v dnf)
```
and use that variable instead.

Reply:
Having it for both would be great!

Reply:
Changes present in v2.